jwst/conftest.py

```diff
@@ -73,3 +74,30 @@ def jail(request, tmpdir_factory):
     os.chdir(str(newpath))
     yield newpath
     os.chdir(old_dir)
+
+@pytest.mark.trylast
+def pytest_configure(config):
+    terminal_reporter = config.pluginmanager.getplugin('terminalreporter')
+    config.pluginmanager.register(TestDescriptionPlugin(terminal_reporter), 'testdescription')
+
+class TestDescriptionPlugin:
+
+    def __init__(self, terminal_reporter):
+        self.terminal_reporter = terminal_reporter
+        self.desc = None
+
+    def pytest_runtest_protocol(self, item):
+        try:
+            self.desc = inspect.getdoc(item.obj)
+        except AttributeError:
+            self.desc = None
+
+    @pytest.hookimpl(hookwrapper=True, tryfirst=True)
+    def pytest_runtest_logstart(self, nodeid, location):
+        if self.terminal_reporter.verbosity == 0:
+            yield
+        else:
+            self.terminal_reporter.write('\n')
+            yield
+            if self.desc:
+                    self.terminal_reporter.write(f'\n{self.desc} ')
```

Comment:
```suggestion
        if self.terminal_reporter.verbosity <= 1:
            yield
        else:
            self.terminal_reporter.write('\n')
            yield
            if self.desc:
                    self.terminal_reporter.write(f'\n{self.desc} ')
```

So that `-v` works as it normally does the default way

```
$ pytest jwst/superbias -v
==================================== test session starts ====================================
platform darwin -- Python 3.8.5, pytest-6.2.1, py-1.10.0, pluggy-0.13.1 -- /Users/jdavies/miniconda3/envs/jwst/bin/python
cachedir: .pytest_cache
rootdir: /Users/jdavies/dev/jwst, configfile: setup.cfg
plugins: asdf-2.7.1, requests-mock-1.8.0, doctestplus-0.8.0, ci-watson-0.5, cov-2.10.1, openfiles-0.5.0
collected 5 items                                                                           

jwst/superbias/tests/test_bias_sub.py::test_basic_superbias_subtraction PASSED        [ 20%]
jwst/superbias/tests/test_bias_sub.py::test_subarray_correction PASSED                [ 40%]
jwst/superbias/tests/test_bias_sub.py::test_dq_propagation PASSED                     [ 60%]
jwst/superbias/tests/test_bias_sub.py::test_nans_in_superbias PASSED                  [ 80%]
jwst/superbias/tests/test_bias_sub.py::test_full_step PASSED                          [100%]

===================================== 5 passed in 3.46s =====================================
```
and `-vv` gives the more verbose output with docstrings.

```
$ pytest jwst/superbias -vv
==================================== test session starts ====================================
platform darwin -- Python 3.8.5, pytest-6.2.1, py-1.10.0, pluggy-0.13.1 -- /Users/jdavies/miniconda3/envs/jwst/bin/python
cachedir: .pytest_cache
rootdir: /Users/jdavies/dev/jwst, configfile: setup.cfg
plugins: asdf-2.7.1, requests-mock-1.8.0, doctestplus-0.8.0, ci-watson-0.5, cov-2.10.1, openfiles-0.5.0
collected 5 items                                                                           


jwst/superbias/tests/test_bias_sub.py::test_basic_superbias_subtraction 
Check basic superbias subtraction. PASSED                                             [ 20%]

jwst/superbias/tests/test_bias_sub.py::test_subarray_correction 
Check that the proper subarray is extracted from the full frame
reference file during subtraction. PASSED                                             [ 40%]

jwst/superbias/tests/test_bias_sub.py::test_dq_propagation 
Check that the PIXELDQ array of the science exposure is correctly
combined with the reference file DQ array. PASSED                                     [ 60%]

jwst/superbias/tests/test_bias_sub.py::test_nans_in_superbias 
Check that no superbias subtraction is done for pixels that have a
value of NaN in the reference file. PASSED                                            [ 80%]

jwst/superbias/tests/test_bias_sub.py::test_full_step 
Test full run of the SuperBiasStep. PASSED                                            [100%]

===================================== 5 passed in 3.39s =====================================
```

Reply:
Ah, okay. I see. Sorry, learning as I go... that works!

Reply:
And the reason for this is that the existing functionality of `pytest -v` is very useful.  You can see how `verbosity is handled in pytest:

https://github.com/pytest-dev/pytest/blob/master/src/_pytest/terminal.py

Reply:
Ohhh okay. Thanks, that's helpful. I don't think I would have known to look at that.

Reply:
Yeah, because their documentation is so poor for this sort of thing, rummaging around through their code to see how they do things is sometimes the most useful way of seeing example implementations.  =)